content/influxdb/v2.0/reference/flux/stdlib/contrib/events/duration.md

```diff
@@ -0,0 +1,171 @@
+---
+title: events.duration() function
+description: >
+  The `events.duration()` function calculates the duration of events.
+  The function determines the time between a record and the subsequent record and
+  associates the duration with the first record (start of the event).
+menu:
+  influxdb_2_0_ref:
+    name: events.duration
+    parent: contrib-events
+weight: 301
+influxdb/v2.0/tags: [functions, events, package]
+related:
+  - /influxdb/v2.0/reference/flux/stdlib/built-in/transformations/elapsed/
+  - /influxdb/v2.0/reference/flux/stdlib/built-in/transformations/stateduration/
+---
+
+The `events.duration()` function calculates the duration of events.
+The function determines the time between a record and the subsequent record and
+associates the duration with the first record (start of the event).
+To calculate the duration of the last event, the function compares the timestamp
+of the final record to the timestamp in the [`stopColumn`](#stopcolumn) or the
+specified [`stop`](#stop) time.
+
+```js
+import "contrib/tomhollingworth/events"
+
+events.duration(
+  unit: 1s,
+  columnName: "duration",
+  timeColumn: "_time",
+  stopColumn: "_stop",
+  stop: 2020-01-01T00:00:00Z
+)
+```
+
+{{% note %}}
+#### Similar functions
+`events.duration()` is similar to `elapsed()` and `stateDuration()`, but differs
+in important ways:
+
+- `elapsed()` drops the first record. `events.duration()` does not.
+- `stateDuration()` calculates the total time spent in a state (determined by a
+  [predicate function](/influxdb/v2.0/reference/glossary/#predicate-function)).
+  `events.duration()` returns the duration between all records and their subsequent records.
+
+For examples, see [below](#compared-to-similar-functions).
+{{% /note %}}
+
+## Parameters
+
+### unit
+<span class="req">Required</span>
```

Comment:
`unit` isn't required, Its default is "1ns"

Reply:
Thanks @tomhollingworth, fixed.